signature/internal/base/envelope.go

```diff
@@ -156,7 +156,7 @@ func validateSignerInfo(info *signature.SignerInfo) error {
 
 	return validateCertificateChain(
 		info.CertificateChain,
-		signingTime,
+		time.Time{},
```

Comment:
Passing a zero time is a security vulnerability. What if line 46 `content.SignerInfo.SignedAttributes.SigningTime` is set to zero by the adversary? It will bypass the check unexpectedly.

It is suggested to change
```go
func validateCertificateChain(certChain []*x509.Certificate, signTime time.Time, expectedAlg signature.Algorithm) error
```
and related functions to
```go
func validateCertificateChain(certChain []*x509.Certificate, signTime *time.Time, expectedAlg signature.Algorithm) error
```
marking `signTime` optional.

Reply:
We have explicit check to not allow zero signing time and its performed both during signing and verification, https://github.com/notaryproject/notation-core-go/blob/199ff3766849257ff3b452d6312b60e7df34b902/signature/internal/base/envelope.go#L167-L169

Also, `content.SignerInfo.SignedAttributes.SigningTime` is asigned parameter and if its modified, the signature will become invalid. Is there any specific security vulnerability you are thinking about?

Reply:
I have concern that we will have regression in the future eventually.